snowflake/exec.go

```diff
@@ -34,19 +36,33 @@ func ExecMulti(db *sql.DB, queries []string) error {
 // QueryRow will run stmt against the db and return the row. We use
 // [DB.Unsafe](https://godoc.org/github.com/jmoiron/sqlx#DB.Unsafe) so that we can scan to structs
 // without worrying about newly introduced columns
-func QueryRow(db *sql.DB, stmt string) *sqlx.Row {
+func QueryRow(ctx context.Context, db *sql.DB, stmt string, args ...interface{}) *sqlx.Row {
 	logrus.Debug("[DEBUG] query stmt ", stmt)
 
 	sdb := sqlx.NewDb(db, "snowflake").Unsafe()
 
-	return sdb.QueryRowx(stmt)
+	if len(args) == 0 {
+		// Don't prepare the statement
+		return sdb.QueryRowxContext(ctx, stmt)
+	}
+
+	// TODO: use this Prepare SQL pattern for the other functions in this file
+	preparedStmt, err := sdb.PreparexContext(ctx, stmt)
+	if err != nil {
+		logrus.Warn(errors.Wrapf(err, "Unable to prepare query (%s)", stmt))
+
+		return nil
+	}
+	defer preparedStmt.Close()
+
+	return preparedStmt.QueryRowxContext(ctx, args)
```

Comment:
This is the only function that has prepare logic built-in right now. Will see if we can add it later with this block as the template